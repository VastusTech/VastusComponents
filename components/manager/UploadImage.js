import React, {useState} from 'react';
import {Button, Icon} from "semantic-ui-react";
import AvatarEditor from "react-avatar-editor";

type Props = {
    imageURL: string,
    callback: (any) => void
};

const onClickSave = (editor, callback) => {
    if (editor) {
        const dataURI = editor.getImageScaledToCanvas().toDataURL("image/jpeg");
        const binary = atob(dataURI.split(',')[1]);
        const array = [];
        for(let i = 0; i < binary.length; i++) {
            array.push(binary.charCodeAt(i));
        }
        callback(new Blob([new Uint8Array(array)], {type: 'image/jpeg'}));
    }
};

const UploadImage = (props: Props) => {
    const [rotation, setRotation] = useState(0);
    const [editorRef, setEditorRef] = useState(null);
    return (
        <div>
            <AvatarEditor
                ref={setEditorRef}
                image={props.imageURL}
                width={250}
                height={250}
                border={50}
                rotate={rotation}
                scale={1.2}
            />
            <Button primary onClick={() => setRotation(p => (p + 90) % 360)}>
                <Icon name="arrow alternate circle right"/>
            </Button>
            <Button primary onClick={() => onClickSave(editorRef, props.callback)}>Upload</Button>
            <Button onClick={() => props.callback(null)}>Cancel</Button>
        </div>
    );
};

export default UploadImage;