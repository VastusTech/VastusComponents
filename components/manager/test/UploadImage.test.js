import React from 'react';
import { shallow } from 'enzyme';
import TestConfig, {store} from "../../../../TestConfig";
import UploadImage from "../UploadImage";

TestConfig();
it('renders without crashing', () => {
    const component = shallow(<UploadImage callback={() => {}} imageURL={null} store={store()}/>);
    expect(component).toMatchSnapshot();
});
