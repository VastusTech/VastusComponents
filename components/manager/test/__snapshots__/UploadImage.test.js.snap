// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UploadImage
    callback={[Function]}
    imageURL={null}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <d
          border={50}
          borderRadius={0}
          className=""
          color={
            Array [
              0,
              0,
              0,
              0.5,
            ]
          }
          disableBoundaryChecks={false}
          height={250}
          image={null}
          onImageChange={[Function]}
          onImageReady={[Function]}
          onLoadFailure={[Function]}
          onLoadSuccess={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onPositionChange={[Function]}
          rotate={0}
          scale={1.2}
          style={Object {}}
          width={250}
        />,
        <Button
          as="button"
          onClick={[Function]}
          primary={true}
          role="button"
        >
          <Icon
            as="i"
            name="arrow alternate circle right"
          />
        </Button>,
        <Button
          as="button"
          onClick={[Function]}
          primary={true}
          role="button"
        >
          Upload
        </Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "border": 50,
          "borderRadius": 0,
          "className": "",
          "color": Array [
            0,
            0,
            0,
            0.5,
          ],
          "disableBoundaryChecks": false,
          "height": 250,
          "image": null,
          "onImageChange": [Function],
          "onImageReady": [Function],
          "onLoadFailure": [Function],
          "onLoadSuccess": [Function],
          "onMouseMove": [Function],
          "onMouseUp": [Function],
          "onPositionChange": [Function],
          "rotate": 0,
          "scale": 1.2,
          "style": Object {},
          "width": 250,
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "children": <Icon
            as="i"
            name="arrow alternate circle right"
          />,
          "onClick": [Function],
          "primary": true,
          "role": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "name": "arrow alternate circle right",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "children": "Upload",
          "onClick": [Function],
          "primary": true,
          "role": "button",
        },
        "ref": null,
        "rendered": "Upload",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <d
            border={50}
            borderRadius={0}
            className=""
            color={
              Array [
                0,
                0,
                0,
                0.5,
              ]
            }
            disableBoundaryChecks={false}
            height={250}
            image={null}
            onImageChange={[Function]}
            onImageReady={[Function]}
            onLoadFailure={[Function]}
            onLoadSuccess={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
            onPositionChange={[Function]}
            rotate={0}
            scale={1.2}
            style={Object {}}
            width={250}
          />,
          <Button
            as="button"
            onClick={[Function]}
            primary={true}
            role="button"
          >
            <Icon
              as="i"
              name="arrow alternate circle right"
            />
          </Button>,
          <Button
            as="button"
            onClick={[Function]}
            primary={true}
            role="button"
          >
            Upload
          </Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "border": 50,
            "borderRadius": 0,
            "className": "",
            "color": Array [
              0,
              0,
              0,
              0.5,
            ],
            "disableBoundaryChecks": false,
            "height": 250,
            "image": null,
            "onImageChange": [Function],
            "onImageReady": [Function],
            "onLoadFailure": [Function],
            "onLoadSuccess": [Function],
            "onMouseMove": [Function],
            "onMouseUp": [Function],
            "onPositionChange": [Function],
            "rotate": 0,
            "scale": 1.2,
            "style": Object {},
            "width": 250,
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": <Icon
              as="i"
              name="arrow alternate circle right"
            />,
            "onClick": [Function],
            "primary": true,
            "role": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "name": "arrow alternate circle right",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": "Upload",
            "onClick": [Function],
            "primary": true,
            "role": "button",
          },
          "ref": null,
          "rendered": "Upload",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
